File: climb2gether - backend/climb2gether - backend/climb2gether - backend/Domain/Post.cs
using Microsoft.AspNetCore.Identity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace climb2gether___backend.Domain
{
    public class Post
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Title { get; set; }
        public string Subtitle { get; set; }
        public string ImgUrl { get; set; }
        public string Content { get; set; }
        public DateTime CreationDate { get; set; }
        [ForeignKey("User")]
        public int UserId { get; set; }
        public User User { get; set; }
        public IEnumerable<PostLikes> PostLikes { get; set; }
    }
}

File: climb2gether - frontend/src/app/_models/Post.ts
import { User } from './user';

export class Post {
    id?: number;
    userId: number;
    userNameSurname?: string;
    content: string;
    img?: File;
    imgURL?: string;
    subtitle: string;
    title: string;
    creationDate: Date;
    user?: User;
    likeCounter?: number;
    postLikedByLoggedUser?: any;
    loggedUserPostLikeId?: number;
}